getconf/base.py

```diff
@@ -70,7 +71,7 @@ def get(self, key, default='', doc=''):
     def getstr(self, key, default='', doc=''):
         """Retrieve a value as a string."""
         assert (
-            default is None or isinstance(default, str)
```

Comment:
This part I do not like.
It would mean that in this case: `STATIC_ROOT = CONFIG.getstr("django.static_root", BASE_DIR / "static")`, that `STATIC_ROOT` will either be a `str` or a  `Path` depending on whether a configuration was provided.

`STATIC_ROOT = CONFIG.getstr("django.static_root", str(BASE_DIR / "static"))` does not seem that bad ?

Reply:
Good point.

Instead, a new `getpath()` could be added. It would return a Path instance, and take a string or a Path as default argument.

Reply:
My first version converted everything to str, but I wasn't comfortable with altering the argument provided.

`str(BASE_DIR / "static")` is actually what I'm using in the meantime, I got used to it.

But `getpath` would be both more explicit and make shorter lines. Going in that direction.

Also reverting any change to list, I can't find any usage to justify the change.